src/devices/vdml_ext_adi.c

```diff
@@ -443,3 +444,64 @@ double ext_adi_potentiometer_get_angle(ext_adi_potentiometer_t potentiometer) {
 	}
 	return_port(smart_port, rtn);
 }
+
+ext_adi_led_t ext_adi_led_init(uint8_t smart_port, uint8_t adi_port) {
+	transform_adi_port(adi_port);
+	claim_port_i(smart_port - 1, E_DEVICE_ADI);
+
+	vexDeviceAdiPortConfigSet(device->device_info, adi_port, E_ADI_DIGITAL_OUT); //digital out according to sample code
+	vexDeviceAdiValueSet(device->device_info, adi_port, (int32_t)LOW);
+	
+	return_port(smart_port - 1, merge_adi_ports(smart_port - 1, adi_port + 1));
+}
+
+int32_t ext_adi_led_set_state(ext_adi_led_t led, bool value) {
+	uint8_t smart_port, adi_port;
+	get_ports(led, smart_port, adi_port);
+	return ext_adi_digital_write(smart_port, adi_port, value);
+}
+
+bool ext_adi_led_get_state(ext_adi_led_t led) {
+	uint8_t smart_port, adi_port;
+	get_ports(led, smart_port, adi_port);
+	return (bool)ext_adi_digital_read(smart_port, adi_port);
+}
+
+int32_t ext_adi_led_set_buffer(ext_adi_led_t led, uint32_t* buffer, uint32_t buffer_length, uint32_t offset) {
+	uint8_t smart_port, adi_port;
+	get_ports(led, smart_port, adi_port);
+	transform_adi_port(adi_port);
+	claim_port_i(smart_port, E_DEVICE_ADI);
+	validate_type(device, adi_port, smart_port - 1, E_ADI_DIGITAL_OUT);
+
+	if(buffer == NULL) {
+        errno = EINVAL;
+        return PROS_ERR;
+    }
+	if (offset < 0) {
+		offset = 0;
+	}
+	else if (offset >= MAX_LED)
+	{
+		offset = MAX_LED - 1;
+	}
+	//TBD: what is the behavior of vexAdiAddrLedSet with a buffer too long (beyond INT_MAX?)
```

Comment:
The buffer is simply a pointer to an address which is why we need to pass in the size (hence why the LED func requires a size), so this is not an issue.